### ISSUE TITLE
"TypeError: an integer is required" when calling function returning double

### ISSUE THREAD

**Issue by belm0** [facebookincubator/cinder#26]
seen as of cinder 5bd8c57 (and it occurs prior to that commit)

```python
from __static__ import double

def test() -> double:
    return double(.5)

test()
```
```
$ python -m compiler --static foo.py
TypeError: an integer is required

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.8/compiler/__main__.py", line 111, in <module>
    exec(codeobj)
  File "/.../foo.py", line 6, in __main__
    test()
SystemError: PyEval_EvalFrameEx returned a result with an error set
```

(Same error even when `test()` is not called from module scope.)

**Comment by czardoz** [comment]
Ah right, returning primitive doubles is not yet supported. I'll make this a compiler reported error, so we don't crash.

**Comment by belm0** [comment]
Our application is heavily numeric, so returning primitive doubles will be key to getting performance out of static modules 🙏

**Comment by czardoz** [comment]
@belm0  This should be fixed since f244f2cd93598c564a4219f7c2b86ac493f2ba7e 

Can you try it out and let me know if you still see an issue? :)

**Comment by czardoz** [comment]
Closing for now, will re-open if necessary
